hazelcast/src/main/java/com/hazelcast/internal/adapter/DataStructureAdapter.java

```diff
@@ -56,4 +56,43 @@
     LocalMapStats getLocalMapStats();
 
     boolean containsKey(K key);
+
+    /**
+     * Contains all methods of {@link DataStructureAdapter}.
+     */
+    enum DataStructureMethods implements DataStructureAdapterMethod {
```

Comment:
This is the only place where we need to add additional information in the future. For every new method in the `DataStructureAdapter` we need to add the signature information here, so the `assumeThatMethodIsAvailable()` method can check for the `@MethodIsUnavailable` annotation.